test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -1864,7 +1864,15 @@ private void postIndexAsyncActions(String[] indices, List<CountDownLatch> inFlig
         }
         while (inFlightAsyncOperations.size() > MAX_IN_FLIGHT_ASYNC_INDEXES) {
             int waitFor = between(0, inFlightAsyncOperations.size() - 1);
-            safeAwait(inFlightAsyncOperations.remove(waitFor));
+            try {
+                assertTrue(
+                    "operation did not complete within timeout",
+                    inFlightAsyncOperations.remove(waitFor).await(60, TimeUnit.SECONDS)
+                );
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                fail(e, "interrupted while waiting for operation to complete");
+            }
```

Comment:
Alternatively, we could limit the number of operations. I think most usages are just a few indexing requests. But decided to keep the change minimal which is a middle ground between the previous code (wait indefinitely) and the current code.